video/input.h

```diff
@@ -98,6 +102,13 @@ class Input
         }
     }
 
+    virtual void readFrameSync(cv::Mat &outFrame)
```

Comment:
I'm not really certain this should be generic method at this level. It only really seems relevant to the Network sources for which it might make more sense to add a 'blocking mode' as that's something that can be done in a much more efficient way.

Reply:
Good point. I've reverted this (I didn't even end up using the function in the end anyway).